jlm/llvm/opt/alias-analyses/Andersen.cpp

```diff
@@ -47,13 +61,21 @@ Andersen::AnalyzeSimpleNode(const rvsdg::simple_node & node)
     AnalyzeConstantAggregateZero(node);
   else if (is<ExtractValue>(op))
     AnalyzeExtractValue(node);
-  else if (is<FreeOperation>(op))
+  else if (is<valist_op>(op))
+    AnalyzeValist(node);
+  else if (is<FreeOperation>(op) || is<ptrcmp_op>(op))
   {
-    // A free does not affect any Points-to-sets
+    // These operations take pointers as input, but do not affect any points-to sets
   }
   else
   {
-    // TODO: Make sure all simple nodes involving pointers are correctly handled by the analysis
+    // This node operation is unknown, make sure it doesn't consume any pointers
+    for (size_t i = 0; i < node.ninputs(); i++)
+    {
+      if (IsOrContainsPointerType(node.input(i)->type()))
+        std::cerr << node.operation().debug_string() << std::endl;
+      JLM_ASSERT(!IsOrContainsPointerType(node.input(i)->type()));
+    }
```

Comment:
This will actually result in some cost in release build. The JLM_ASSERT is only enabled in a debug build (-DJLM_ENABLE_ASSERT), but not in a release build. If you also print the debug string, then the loop does not only contain noops, and the compiler won't be able to optimize it away. Thus, you will actually iterate through all inputs for every single node that is not handled by the analysis (integer operations, float operations, etc.) in optimized code. This is a high price to pay for the convenience of the operation name that you could also get by firing up the debugger (when it is needed).

Reply:
I am sorry, but I did not see this the first time looking at the PR.

Reply:
I completely understand, good thing you caught it. It was only there to quickly get the missing operations, and should have had `TODO Remove` plastered all over it.